src/CountryData.js

```diff
@@ -77,15 +78,21 @@ function initCountries(prefix, defaultMask, alwaysDefaultMask) {
 
 export default class CountryData {
   constructor (
-    enableAreaCodes, regions,
+    enableAreaCodes, enableTerritories, regions,
     onlyCountries, preferredCountries, excludeCountries, preserveOrder,
     localization, masks, areaCodes,
     prefix, defaultMask, alwaysDefaultMask,
     priority,
   ) {
     let filteredCountries = enableAreaCodes ?
-      initCountriesAndAreaCodes(enableAreaCodes, prefix, defaultMask, alwaysDefaultMask) :
-      initCountries(prefix, defaultMask, alwaysDefaultMask);
+      initCountriesAndAreaCodes(rawCountries, enableAreaCodes, prefix, defaultMask, alwaysDefaultMask) :
+      initCountries(rawCountries, prefix, defaultMask, alwaysDefaultMask);
+    if (enableTerritories) {
+      let filteredTerritories = enableAreaCodes ?
+        initCountriesAndAreaCodes(rawTerritories, enableAreaCodes, prefix, defaultMask, alwaysDefaultMask) :
+        initCountries(rawTerritories, prefix, defaultMask, alwaysDefaultMask);
+      filteredCountries = this.initTerritories(filteredTerritories, filteredCountries);
+    }
```

Comment:
Seems I previously incorrectly suggested to place regions filter above territories initialization. Thanks for proper placement!